src/databricks/labs/ucx/recon/_base.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class TableDescriptor:
+    catalog: str
+    schema: str
+    table: str
+
+
+@dataclass
+class ColumnMetadata:
+    name: str
+    data_type: str
+
+
+@dataclass
+class TableMetadata:
+    descriptor: TableDescriptor
+    columns: list[ColumnMetadata]
+
+    def get_column_metadata(self, column_name: str) -> ColumnMetadata | None:
+        for column in self.columns:
+            if column.name == column_name:
+                return column
+        return None
+
+
+@dataclass
+class DataProfilingResult:
+    row_count: int
+    table_metadata: TableMetadata
+
+
+@dataclass
+class SchemaComparisonEntry:
```

Comment:
you could reuse/extend https://github.com/databrickslabs/ucx/blob/953c4837614b355d03bac9c0f201ca2a39e11a14/src/databricks/labs/ucx/hive_metastore/mapping.py#L23-L29
or 
https://github.com/databrickslabs/ucx/blob/953c4837614b355d03bac9c0f201ca2a39e11a14/src/databricks/labs/ucx/hive_metastore/mapping.py#L63-L65